src/main/java/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -294,35 +298,33 @@ private static void updateMapUrlAndSizeLabel(final DownloadFileDescription map,
   }
 
   private static String createLabelText(final MapAction action, final DownloadFileDescription map) {
-    final String DOUBLE_SPACE = "&nbsp;&nbsp;";
+    final String doubleSpace = "&nbsp;&nbsp;";
 
-    final long mapSize;
-    String labelText = "<html>" + map.getMapName() + DOUBLE_SPACE + " v" + map.getVersion();
+    final StringBuilder sb = new StringBuilder();
+    sb.append("<html>" + map.getMapName() + doubleSpace + " v" + map.getVersion());
 
+    final Optional<Long> mapSize;
```

Comment:
I changed `mapSize` from `long` to `Optional<Long>` here to get rid of the smell of using the magic `-1` constant.  Since we were using `Optional<Long>` already, it seemed like a no-brainer.